File: 0189_Rotate Array/RotateArray.java
 class Solution {
   public void rotate(int[] nums, int k) {		
        int n = nums.length;
        k = k % n;
        if(k <= 0) return;
        reverse(nums, 0, nums.length - 1);
        reverse(nums, 0, k - 1);
        reverse(nums, k, nums.length - 1);
	}
	    
    private void reverse(int[] nums, int l, int r){
        while(l < r){
            int tmp = nums[l];
            nums[l++] = nums[r];
            nums[r--] = tmp;
            
        }
    }
}

File: 0189_Rotate Array/RotateArray.cs
public class Solution {
    public void Rotate(int[] nums, int k) {
        var n = nums.Length;
        k %= n;
        Reverse(nums, 0, n-1);
        Reverse(nums, 0, k-1);
        Reverse(nums, k, n-1);
    }
    
    private void Reverse(int[] nums, int l , int r)
    {
        while(l < r)
        {
            var tmp = nums[l];
            nums[l++] = nums[r];
            nums[r--] = tmp;
        }
    }
}